python/test/unit/geometry/test_bounding_box_tree.py

```diff
@@ -496,3 +497,82 @@ def test_surface_bbtree_collision(dtype):
 
     collisions = compute_collisions_trees(bbtree1, bbtree2)
     assert len(collisions) == 1
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_determine_point_ownership(dtype):
+    """Find point owners (ranks and cells) using bounding box trees + global communication
+    and compare to point ownership data results."""
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+
+    tdim = 2
+    num_cells_side = 4
+    h = dtype(1.0 / num_cells_side)
+    mesh = create_unit_square(MPI.COMM_WORLD,
+                              num_cells_side, num_cells_side,
+                              CellType.quadrilateral, dtype=dtype)
+    cell_map = mesh.topology.index_map(tdim)
+
+    def test_po_against_bb_tree(mesh,points,cells):
+        cells = np.array([cell for cell in cells if cell < cell_map.size_local])
+        num_points = points.shape[0]
+        po = determine_point_ownership(mesh,points,cells)
+        tree = bb_tree(mesh, mesh.topology.dim, cells)
+        assert(mesh.topology.cell_type in [CellType.quadrilateral,CellType.hexahedron])
```

Comment:
Is this assertion needed? I get that the test only makes sense if the grid is axis aligned with bounding boxes, but I would rather have a comment regarding this at mesh creation.